Fixed issues in Drizzle and React tutorial reported in Issue #1074. (#1103)
trufflesuite/trufflesuite.com: src/guides/getting-started-with-drizzle-and-react/index.md

  },
};

// setup drizzle
const drizzle = new Drizzle(options);

ReactDOM.render(
  <React.StrictMode>
    <App drizzle={drizzle} />
  </React.StrictMode>,
  document.getElementById('root')
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals

> reportWebVitals();

Author: This is actually the boilerplate code from create-react-app. I can't claim credit for this :)

Replace the contents of `client/src/App.js` with the following:

```javascript
import React from 'react';

class App extends React.Component {

  state = { loading: true, drizzleState: null };

  componentDidMount() {
    const { drizzle } = this.props;

    // subscribe to changes in the store
    this.unsubscribe = drizzle.store.subscribe(() => {

      // every time the store updates, grab the state from drizzle
      const drizzleState = drizzle.store.getState();

      // check to see if it's ready, if so, update local component state
      if (drizzleState.drizzleStatus.initialized) {
        this.setState({ loading: false, drizzleState });
      }
    });
  }

  componentWillUnmount() {
    this.unsubscribe();
  }

  render() {
    if (this.state.loading) return "Loading Drizzle...";
    return <div className="App">Drizzle is ready</div>;
  }
}
```

#### Adding state variables

> The first thing we have done is to set up our state variables inside our App component:

Reviewer: ```suggestion
First, we set up our state variables inside our App component:
```

```javascript
state = { loading: true, drizzleState: null };
```

> We are using two state variables here:

Reviewer: Come to think of it may as well remove the passive voice here. 

```suggestion
We use two state variables here:
```

1. `loading` — Indicates if Drizzle has finished initializing and the app is ready. The initialization process includes instantiating `web3` and our smart contracts, fetching any available Ethereum accounts and listening (or, in cases where subscriptions are not supported: polling) for new blocks.
2. `drizzleState` — This is where we will store the state of the Drizzle store in our top-level component. If we can keep this state variable up-to-date, then we can simply use simple `props` and `state` to work with Drizzle (i.e. you don't have to use any Redux or advanced React patterns).

#### Adding some initialization logic

> Next we have provided the `componentDidMount` lifecycle method for the component. The `componentDidMount` lifecycle method will run after the component output has been rendered to the DOM. This allows us to run some initialization logic.

Reviewer: ```suggestion
Next we provide the `componentDidMount` lifecycle method for the component. The `componentDidMount` lifecycle method runs after the component output is rendered to the DOM. This allows us to run some initialization logic.
```

#### Replace the `render` method

> Finally, we have replaced the boilerplate render method with something that applies to us better:

Reviewer: ```suggestion
Finally, we replace the boilerplate render method with something that applies to us better:
```

In the next section, we will replace "Drizzle is ready" with an actual component that will read from the store. If you refresh your browser and run this app now, you should see "Loading Drizzle..." briefly flash on screen and then subsequently "Drizzle is ready".

> ### Full component code

Reviewer: I may be missing context since I'm looking at this in GitHub, but shouldn't we leave in the section with what the completed code should look like?

Author: The full source of the App.js is now higher in the tutorial under the heading 'Wire up the App component'. The issue that you reported noted the difference in the App.js between the code in the tutorial and the code that you had locally from running create-react-app. By default create-react-app creates a React functional component rather than the class shown in the tutorial. A class component is preferred for managing state and using life cycle methods. You can do it in a functional component but, there are some extra hoops to jump through first. The approach I took was to instruct the user to replace the content of App.js with the code presented in the tutorial - that is, the full source code for the component. The tutorial then explains the App.js code. We could add the full source for App.js at the end - but there is only a few additional lines. I thought that would be unnecessary as its pretty clear from the tutorial what's happening. However, I am happy to add it if you think it's worthwhile.

Reviewer: Nope, that seems fine! I figured I was missing some context, thank you for clarifying!
